editoast/src/infra_cache/operation/mod.rs

```diff
@@ -16,23 +17,93 @@ pub use self::delete::DeleteOperation;
 use crate::error::Result;
 use crate::infra_cache::ObjectCache;
 use crate::modelsv2::DbConnection;
-use editoast_schemas::primitives::OSRDObject as _;
+use editoast_schemas::infra::InfraObject;
 use editoast_schemas::primitives::ObjectRef;
+use editoast_schemas::primitives::ObjectType;
 
-editoast_common::schemas! { Operation, }
+editoast_common::schemas! {
+    Operation,
+    &json_patch::AddOperation,
+    &json_patch::CopyOperation,
+    &json_patch::MoveOperation,
+    &json_patch::PatchOperation,
+    &json_patch::RemoveOperation,
+    &json_patch::ReplaceOperation,
+    &json_patch::TestOperation,
+    &json_patch::Patch,
+}
 
-#[derive(Clone, Debug, Deserialize, Serialize, PartialEq, ToSchema)]
-#[serde(tag = "operation_type", deny_unknown_fields)]
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq)]
+#[serde(tag = "operation_type", deny_unknown_fields, rename_all = "UPPERCASE")]
 pub enum Operation {
-    #[serde(rename = "CREATE")]
-    #[schema(value_type = RailjsonObject)]
-    Create(Box<RailjsonObject>),
-    #[serde(rename = "UPDATE")]
+    Create(Box<InfraObject>),
     Update(UpdateOperation),
-    #[serde(rename = "DELETE")]
     Delete(DeleteOperation),
 }
 
+// We implement ToSchema this way because:
+//
+// - The crate `json_patch` kindly provides us schema definitions for the JSON Patch standard :)
+// - However the names of the subcomponents are quite generic (AddOperation, CopyOperation, etc.)
+// - That overlaps with our own `UpdateOperation` and `DeleteOperation` types (especially `RemoveOperation`)
+// - We also can't `#[schema(inline)]` the content of the variants because utoipa doesn't allow it :'(
+// - So we have to manually define the schema for the `Operation` enum in order not to break the front too much
```

Comment:
Yeah because the name is too similar, but the purpose is very different.
The comment isn't clear enough, I'll change it. What I meant was that the frontend relies a lot on the `Operation` schema, so changing its structure to allow a generated implementation of `ToSchema` is not a good idea.

Reply:
Done in https://github.com/OpenRailAssociation/osrd/pull/7553/commits/e5e69028ab3f93e1358d999c22d064c91087cfc6